src/main/java/BankingSystem.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    private void addIncome() {
+        System.out.println("Enter income:");
+
+        try {
+            int income = Integer.parseInt(scanner.next());
+
+            if (income < 0 || income == 0) {
+                System.out.println("Money can't be a zero or a negative number!\n");
+                return;
+            }
+
+            updateAccount(currentAccount.getCard().number(), income);
+            System.out.println("Income was added!\n");
+        } catch (NumberFormatException e) {
+            System.out.println("You should enter the amount of money you want to add to your balance!\n");
+            addIncome();
+        }
+    }
+
+    private void doTransfer() {
+        System.out.println("Transfer\n" +
+                           "Enter card number:");
+        String recipientCardNumber = scanner.next();
+
+        if (recipientCardNumber.equals(currentAccount.getCard().number())) {
+            System.out.println("You can't transfer money to the same account!\n");
+            return;
+        }
+
+        if (!cardNumberValidation(recipientCardNumber)) {
+            System.out.println("Probably you made a mistake in the card number. Please try again!\n");
+            return;
+        }
+
+        if (!dao.contains(recipientCardNumber)) {
+            System.out.println("Such a card does not exist.\n");
+            return;
+        }
+
+        System.out.println("Enter how much money you want to transfer:");
+
+        try {
+            int moneyToTransfer = Integer.parseInt(scanner.next());
+
+            if (moneyToTransfer < 0 || moneyToTransfer == 0) {
+                System.out.println("Money can't be a zero or a negative number!\n");
+                return;
+            }
+
+            if (moneyToTransfer > currentAccount.getBalance()) {
+                System.out.println("Not enough money!\n");
+                return;
+            }
+
+            dao.update(recipientCardNumber, moneyToTransfer);
+            updateAccount(currentAccount.getCard().number(), -moneyToTransfer);
+            System.out.println("Success!\n");
+        } catch (NumberFormatException e) {
+            System.out.println("You should enter the amount of money you want to transfer!\n");
+            doTransfer();
+        }
+    }
+
+    private void closeAccount() {
+        dao.delete(currentAccount);
+        System.out.println("The account has been closed!\n");
+        currentAccount = null;
+    }
+
+    private void logOut() {
+        currentAccount = null;
+        System.out.println("You have successfully logged out!\n");
+    }
+
+    private void updateAccount(String cardNumber, int transferredMoney) {
+        dao.update(cardNumber, transferredMoney);
+        dao.get(cardNumber, currentAccount.getCard().pin())
+                .ifPresent(account -> currentAccount.setBalance(currentAccount.getBalance() + transferredMoney));
+    }
+
+    public boolean cardNumberValidation(String cardNumber) {
```

Comment:
Can you use the private access modifier?

Reply:
Done